packages/discord.js/src/structures/Presence.js

```diff
@@ -254,7 +254,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        this.emoji?.id === activity.emoji?.id)
```

Comment:
This won't fix the overall problem, if the user changes their custom status from one unicode emoji to another then it won't fire the event either, so both the `emoji?.name` and `emoji?.id` should be checked

Reply:
^ exactly

Reply:
you're right... wonder why it worked for me in testing then.

however, what if the emoji's name is updated, won't that trigger a false update?

Reply:
Your can't prevent that

Reply:
Alright I'll just check for `name` & `id` then 👍

Reply:
> Alright I'll just check for `name` & `id` then 👍

Better yet, might as well properly check for it: `this.emoji?.id ? this.emoji.id === activity.emoji?.id : this.emoji?name === activity.emoji?.name`

Reply:
idi

Reply:
> idk

Well I've tested it, it seems to work!